index.bs

```diff
@@ -3405,6 +3420,56 @@ and |session|:
 
 </div>
 
+<div algorithm="locate nodes using accessibility attributes">
+To <dfn>locate nodes using accessibility attributes</dfn> with given |context nodes|, |selector|,
+|maximum returned node count|, |session|, and |returned nodes|:
+
+1. If |selector|'s <code>role</code> and |selector|'s <code>name</code> are missing, return [=error=] with [=error code=]
+   [=invalid selector=].
+
+1. If |returned nodes| is null:
+
+  1. Set |returned nodes| to an empty [=/list=].
+
+1. If |maximum returned node count| is not null and [=list/size=] of |returned
+   nodes| is greater or equal to |maximum returned node count|:
+
+   1. Return [=success=] with data null.
+
+1. For each |context node| in |context nodes|:
+
+  1. Let |match| be true.
+
+  1. If |selector|'s <code>role</code> is present:
+
+    1. Let |role| be the result of computing the [=accessible role=] of |context node|.
+
+    1. If |selector|'s <code>role</code> [=is|is not=] |role|:
+
+      1. Set |match| to false.
+
+  1. If |selector|'s <code>name</code> is present:
+
+    1. Let |name| be the result of computing the [=accessible name=] of |context node|.
+
+    1. If |selector|'s <code>name</code> [=is|is not=] |name|:
+
+      1. Set |match| to false.
+
+  1. If |match| is true:
+
+    1. [=list/Append=] |context node| to |returned nodes|.
```

Comment:
So we can end up with more than `|maximum returned node count|` nodes being returned? That seems to violate the assert in the calling steps. I think we either need to check each time we append, or truncate the list at the end.